Sources/Sentry/SentryScreenshot.m

```diff
@@ -38,6 +45,27 @@ @implementation SentryScreenshot
     return result;
 }
 
+- (void)saveScreenShots:(NSString *)path
+{
+    NSArray<UIWindow *> *windows = [SentryDependencyContainer.sharedInstance.application windows];
+
+    for (UIWindow *window in windows) {
+        UIGraphicsBeginImageContext(window.frame.size);
+
+        int index = 0;
+        if ([window drawViewHierarchyInRect:window.bounds afterScreenUpdates:false]) {
+            UIImage *img = UIGraphicsGetImageFromCurrentImageContext();
+            NSString *name = index == 0
+                ? @"screenshot.png"
+                : [NSString stringWithFormat:@"screenshot-%i.png", index++ + 1];
+            NSString *fileName = [path stringByAppendingPathComponent:name];
+            [UIImagePNGRepresentation(img) writeToFile:fileName atomically:YES];
+        }
+
+        UIGraphicsEndImageContext();
```

Comment:
`m`: Shouldn't we make sure that we call this on the main thread as well?

Reply:
We need to run this on its own thread to avoid lock. The main thread will be blocked if the crash occurred on it.

Reply:
Does this mean we spin off a new thread when crashing? If yes, I'm unsure if this is the best thing to do, as most likely it will fail?

Reply:
We don't. We are using the signal handling thread which is not the main thread, I believe is just the way things works with signal handling. I tried to queue in the main thread and the app froze.

Reply:
Thanks for explaining that. Can you please add this vital insight as a comment?